src/Plugin/migrate/source/YAML.php

```diff
@@ -0,0 +1,175 @@
+<?php
+
+namespace Drupal\migrate_default_content\Plugin\migrate\source;
+
+use Drupal\migrate\MigrateException;
+use Drupal\migrate\Plugin\migrate\source\SourcePluginBase;
+use Drupal\migrate\Plugin\MigrationInterface;
+
+/**
+ * Source for YAML.
+ *
+ * If the YAML file contains non-ASCII characters, make sure it includes a
+ * UTF BOM (Byte Order Marker) so they are interpreted correctly.
+ *
+ * @MigrateSource(
+ *   id = "yaml"
```

Comment:
I feel this file can be a separate module like migrate_source_csv. If it's too specific for migrate_default_content we should avoid this id in case it clashes with another source plugin.